app/views/nonprofits/edit.html.erb

```diff
@@ -23,8 +23,31 @@
         <%= f.phone_field :phone, class: "form-control", placeholder: "Update phone number" %>
       </div>
       
+      <% @nonprofit.social_profile ||= SocialProfile.new %>
+      <%=f.fields_for :social_profile do |social_profile_form| %>
```

Comment:
@ianroberts131 this issue is an ongoing problem for developers, and a blog post about it would be worthwhile 👍 

Regarding the chosen solution, its clear and gets the job done. There are alternatives that merit considering in this kind of situation. The alternatives are:

Option 1. Moving the behaviour into `NonprofitsController#edit` to initialize `@nonprofit.social_profile`

```rb
class NonprofitsController < ApplicationController
  ...
  def edit
    @nonprofit = current_user.nonprofits.find(params[:id])

    # The "build_social_profile" method is generated by the `has_one :social_profile` association
    # and is similar to `@nonprofit.social_profile = SocialProfile.new(sociable: @nonprofit)`
    @nonprofit.build_social_profile if @nonprofilt.social_profile.nil?
  end
```

Option 2. Overriding the `Nonprofit#social_profile` auto-generated method provided by `ActiveRecord::Base` (the superclass of `Nonprofit`:

```rb
class Nonprofit < ActiveRecord::Base
  ...
  def social_profile
    # super calls the social_profile method provided by
    # superclass ActiveRecord::Base, if super returns nil,
    # then the social_profile is initialized by `build_social_profile`
    super || build_social_profile
  end
```

We can discuss today the relative merits of the alternatives.